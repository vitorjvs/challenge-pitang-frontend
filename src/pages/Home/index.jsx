import React from 'react';
import Page from '../../components/Page';
import VaccineForm from '../../components/Forms/Form';

export default function Home() {
  return (
    <Page title="Olá, seja bem-vindo! Preencha as informações abaixo para confirmar o seu agendamento da vacinação contra a COVID-19.">
      <VaccineForm />
    </Page>
  );
}
