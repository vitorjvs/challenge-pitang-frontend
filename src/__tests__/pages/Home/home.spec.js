/* eslint-disable no-undef */
import React from 'react';
import { render, screen } from '@testing-library/react';
import Home from '../../../pages/Home';

describe('Home Page', () => {
  it('must render', () => {
    const { asFragment } = render(<Home />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('Must contains all three forms labels', () => {
    render(<Home />);
    const formsFields = ['Nome', 'Data de nascimento', 'Data do agendamento'];
    formsFields.map((fieldName) => expect(screen.getByText(fieldName)).toBeInTheDocument());
  });

  it('Must contains a submit button', () => {
    render(<Home />);
    const submitButton = screen.getByRole('button', {
      name: 'Confirmar Agendamento',
    });
    expect(submitButton).toBeInTheDocument();
  });
});
