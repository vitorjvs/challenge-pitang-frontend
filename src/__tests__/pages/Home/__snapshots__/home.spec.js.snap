// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page must render 1`] = `
<DocumentFragment>
  <div
    class="mt-5 container-fluid"
  >
    <div
      class="card"
    >
      <div
        class="card-header"
      >
        <div
          class="card-title h5"
        >
          <center>
            Olá, seja bem-vindo! Preencha as informações abaixo para confirmar o seu agendamento da vacinação contra a COVID-19.
          </center>
        </div>
      </div>
      <div
        class="card-body"
      >
        <form
          action="#"
          class="mt-5"
        >
          <div
            class="container"
          >
            <div
              class="justify-content-md-center row"
            >
              <div
                class="col-md-3"
              >
                <label
                  for="name"
                >
                  <b>
                    Nome
                  </b>
                </label>
              </div>
              <div
                class="col"
              >
                <input
                  class="text-input"
                  name="name"
                  placeholder="Insira seu nome"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
          <br />
          <div
            class="container"
          >
            <div
              class="justify-content-md-center row"
            >
              <div
                class="col-md-3"
              >
                <label
                  for="birthDate"
                >
                  <b>
                    Data de nascimento
                  </b>
                </label>
              </div>
              <div
                class="col"
              >
                <div
                  class="react-datepicker-wrapper"
                >
                  <div
                    class="react-datepicker__input-container"
                  >
                    <input
                      class=""
                      name="birthDate"
                      placeholder="Insira sua data de nascimento"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <br />
          <div
            class="container"
          >
            <div
              class="justify-content-md-center row"
            >
              <div
                class="col-md-3"
              >
                <label
                  for="appointmentTime"
                >
                  <b>
                    Data do agendamento
                  </b>
                </label>
              </div>
              <div
                class="col"
              >
                <div
                  class="react-datepicker-wrapper"
                >
                  <div
                    class="react-datepicker__input-container"
                  >
                    <input
                      class=""
                      name="appointmentTime"
                      placeholder="Selecione o dia e horário do seu agendamento"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <button
            class="mt-5 text-center float-right btn btn-primary"
            type="submit"
          >
            Confirmar Agendamento
          </button>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
